parquet-column/src/main/java/org/apache/parquet/column/impl/ColumnWriterBase.java

```diff
@@ -64,6 +72,38 @@
     this.repetitionLevelColumn = createRLWriter(props, path);
     this.definitionLevelColumn = createDLWriter(props, path);
     this.dataColumn = props.newValuesWriter(path);
+    this.bloomFilter = null;
+    this.bloomFilterWriter = null;
+  }
+
+  ColumnWriterBase(
+    ColumnDescriptor path,
+    PageWriter pageWriter,
+    BloomFilterWriter bloomFilterWriter,
+    ParquetProperties props
+  ) {
+    this(path, pageWriter, props);
```

Comment:
Travis fails because `bloomFilter` and `bloomFilterWriter` gets value twice (once in the other constructor because of this call and once in this constructor) while they are final.
I suggest moving all the code from the other constructor to this one so the initialization would be done only once. Then, you won't invoke the other constructor from this one but the other way around. The other constructor would only call this one (with `bloomFilterWriter = null`) and nothing else.

Reply:
Fixed, thanks.